be/src/exec/schema_scanner/schema_table_options_scanner.cpp

```diff
@@ -0,0 +1,134 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#include "exec/schema_scanner/schema_table_options_scanner.h"
+
+#include "runtime/client_cache.h"
+#include "runtime/exec_env.h"
+#include "runtime/runtime_state.h"
+#include "util/thrift_rpc_helper.h"
+#include "vec/common/string_ref.h"
+#include "vec/core/block.h"
+#include "vec/data_types/data_type_factory.hpp"
+
+namespace doris {
+std::vector<SchemaScanner::ColumnDesc> SchemaTableOptionsScanner::_s_tbls_columns = {
+        {"TABLE_NAME", TYPE_VARCHAR, sizeof(StringRef), true},
+        {"TABLE_CATALOG", TYPE_VARCHAR, sizeof(StringRef), true},
+        {"TABLE_SCHEMA", TYPE_VARCHAR, sizeof(StringRef), true},
+        {"TABLE_MODEL", TYPE_STRING, sizeof(StringRef), true},
+        {"TABLE_MODEL_KEY", TYPE_STRING, sizeof(StringRef), true},
+        {"DISTRIBUTE_KEY", TYPE_STRING, sizeof(StringRef), true},
+        {"DISTRIBUTE_TYPE", TYPE_STRING, sizeof(StringRef), true},
+        {"BUCKETS_NUM", TYPE_INT, sizeof(int32_t), true},
+        {"PARTITION_NUM", TYPE_INT, sizeof(int32_t), true},
+        {"PROPERTIES", TYPE_STRING, sizeof(StringRef), true},
+};
+
+SchemaTableOptionsScanner::SchemaTableOptionsScanner()
+        : SchemaScanner(_s_tbls_columns, TSchemaTableType::SCH_TABLE_OPTIONS) {}
+
+Status SchemaTableOptionsScanner::start(RuntimeState* state) {
+    _block_rows_limit = state->batch_size();
+    _rpc_timeout_ms = state->execution_timeout() * 1000;
+    return Status::OK();
+}
+
+Status SchemaTableOptionsScanner::get_block_from_fe() {
+    TNetworkAddress master_addr = ExecEnv::GetInstance()->master_info()->network_address;
+
+    TSchemaTableRequestParams schema_table_request_params;
+    for (int i = 0; i < _s_tbls_columns.size(); i++) {
```

Comment:
We may need to add a version mechanism to control if the columns in `_s_tbls_columns` changed in the future, which keeps FE and BE seeing a consistent "table header"

Reply:
I think this is  handled using TSchemaTableRequestParams.
TSchemaTableRequestParams BE will set the BE column names in a list and send to FE. FE handle request in MetadataGenerator and filters out the columns in function filterColumns and fills according to request from BE. so even some column count different between FE and BE, this method would solve it.